### PR TITLE
fix

### LINKED ISSUE #38: Traceback on prolog-epilog relation-changed hook
```bash
Traceback (most recent call last):
  File "./src/charm.py", line 356, in <module>
    main(SlurmConfiguratorCharm)
  File "/var/lib/juju/agents/unit-slurm-configurator-0/charm/venv/ops/main.py", line 401, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-slurm-configurator-0/charm/venv/ops/main.py", line 140, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-slurm-configurator-0/charm/venv/ops/framework.py", line 234, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-slurm-configurator-0/charm/venv/ops/framework.py", line 678, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-slurm-configurator-0/charm/venv/ops/framework.py", line 723, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-slurm-configurator-0/charm/src/interface_prolog_epilog.py", line 74, in _on_relation_changed
    self._charm.set_slurm_license_available(True)
AttributeError: 'SlurmConfiguratorCharm' object has no attribute 'set_slurm_license_available'
unit-slurm-configurator-0: 20:20:24 ERROR juju.worker.uniter.operation hook "prolog-epilog-relation-changed" (via hook dispatching script: dispatch) failed: exit status 1
unit-slurm-configurator-0: 20:20:24 INFO juju.worker.uniter awaiting error resolution for "relation-changed" hook
```

### PR DESCRIPTION
**Pull request by ducks23**
Fixes https://github.com/omnivector-solutions/slurm-charms/issues/38

### CODE REVIEW

File: charm-slurm-configurator/src/interface_prolog_epilog.py [hunk 1]
```diff
@@ -70,12 +70,10 @@ def _on_relation_changed(self, event):
             'slurmctld_prolog_path': prolog
         })
 
-        # Store the license path in the charm's state
-        self._charm.set_slurm_license_available(True)
         self.on.prolog_epilog_available.emit()
 
     def _on_relation_broken(self, event):
-        self._charm.set_slurm_license_available(False)
+        self._stored.prolog_epilog = None
```
Comment (jamesbeedy) [inline charm-slurm-configurator/src/interface_prolog_epilog.py]: @ducks23 we should try and be consistent with our typing here. `self._stored.prolog_epilog` is initialized to a dict in `__init__()`, set to a string in `relation_changed`, and set to `NoneType` in `broken`. As part of this pr, will you ensure that the type matches all the way across.

**Review by jamesbeedy — CHANGES_REQUESTED** [review]
suggestion on type consistency

File: charm-slurm-configurator/src/interface_prolog_epilog.py (hunk 1 above)
Comment (ducks23) [reply charm-slurm-configurator/src/interface_prolog_epilog.py]: it is set to a dict in relation changed

**Comment by jamesbeedy** [comment]
👍

**Review by jamesbeedy — APPROVED** [review]

**Pull request merged**